### PR TITLE
AccessManager: Remove classes

### PR DESCRIPTION
**Pull request by Amxx**
Fixes LIB-1055
Fixes LIB-1045 (part)

We are missing:
- [ ] Tests for the setContractMode
- [ ] Tests for the contract delay
- [ ] Renaming group → role ?

#### PR Checklist

- [ ] Tests
- [ ] Documentation
- [ ] Changeset entry (run `npx changeset add`)

### CODE REVIEW

File: contracts/access/manager/AccessManager.sol [hunk 1]
```diff
@@ -52,7 +52,8 @@ contract AccessManager is Context, Multicall, IAccessManager {
     using Time for *;
 
     struct AccessMode {
```
Comment (Amxx) [inline contracts/access/manager/AccessManager.sol:54]: Should we rename that structure ?

Comment (frangio) [reply contracts/access/manager/AccessManager.sol:54]: Yes! This name was left over from when it was the AccessMode enum :sweat_smile:

Comment (frangio) [reply contracts/access/manager/AccessManager.sol:54]: `ContractConfig` or something?

File: contracts/access/manager/AccessManager.sol [hunk 2]
```diff
@@ -800,41 +754,51 @@ contract AccessManager is Context, Multicall, IAccessManager {
...
-            selector == this.setContractClosed.selector
+            selector == this.setContractAdminDelay.selector
         ) {
-            // Restricted to ADMIN with no delay beside any execution delay the caller may have
             return (true, ADMIN_GROUP, 0);
-        } else if (selector == this.grantGroup.selector || selector == this.revokeGroup.selector) {
-            // First argument is a groupId. Restricted to that group's admin with no delay beside any execution delay the caller may have.
+        }
+
+        // Restricted to ADMIN with the class delay corresponding to the target
```
Comment (Amxx) [inline contracts/access/manager/AccessManager.sol:781]:
```suggestion
        // Restricted to ADMIN with the delay corresponding to the target
```

Comment (frangio) [inline contracts/access/manager/AccessManager.sol]:
```suggestion
        // Restricted to ADMIN with the admin delay corresponding to the target
```

File: contracts/access/manager/AccessManager.sol [hunk 3]
```diff
@@ -800,41 +754,51 @@ contract AccessManager is Context, Multicall, IAccessManager {
...
-            selector == this.setContractClosed.selector
+            selector == this.setContractAdminDelay.selector
         ) {
-            // Restricted to ADMIN with no delay beside any execution delay the caller may have
             return (true, ADMIN_GROUP, 0);
-        } else if (selector == this.grantGroup.selector || selector == this.revokeGroup.selector) {
-            // First argument is a groupId. Restricted to that group's admin with no delay beside any execution delay the caller may have.
+        }
+
+        // Restricted to ADMIN with the delay corresponding to the target
```
Comment (Amxx) [inline contracts/access/manager/AccessManager.sol:781]:
```suggestion
        // Restricted to ADMIN with the admin delay corresponding to the target
```

File: contracts/access/manager/AccessManager.sol (hunk 1 above)
Comment (Amxx) [reply contracts/access/manager/AccessManager.sol:54]: I did that, but I'd like to consider replacing "Contract" with "Target".

- `setContractFunctionGroup` → `setTargetFunctionGroup`
- `isContractClosed` → `isTargetClosed`
...

WDYT?

Comment (frangio) [reply contracts/access/manager/AccessManager.sol]: Definitely, I like that.

File: .changeset/wild-peas-remain.md
[the remark this answers is not in the document]
Comment (frangio) [reply .changeset/wild-peas-remain.md:1]: I think we need to merge `master` into `feat/access-manager`, then merge `feat/access-manager` into `audit/wip/2a-2b`.

[the remark this answers is not in the document]
Comment (Amxx) [reply .changeset/wild-peas-remain.md:1]: Done 
I had to resolve 2 conflicts, one for each merge. Maybe you should check.

File: contracts/access/manager/AccessManager.sol [hunk 4]
```diff
@@ -75,26 +76,19 @@ contract AccessManager is Context, Multicall, IAccessManager {
...
+        uint48 timepoint;
+        uint32 nonce;
     }
 
     uint64 public constant ADMIN_GROUP = type(uint64).min; // 0
     uint64 public constant PUBLIC_GROUP = type(uint64).max; // 2**64-1
 
-    mapping(address target => AccessMode mode) private _contractMode;
-    mapping(uint64 classId => Class) private _classes;
+    mapping(address target => TargetConfig mode) private _contracts;
```
Comment (frangio) [inline contracts/access/manager/AccessManager.sol:90]: Rename to `_targets`?

File: contracts/access/manager/AccessManager.sol [hunk 5]
```diff
@@ -441,8 +433,8 @@ contract AccessManager is Context, Multicall, IAccessManager {
             revert AccessManagerLockedGroup(groupId);
         }
 
-        (Time.Delay updated, uint48 effect) = _groups[groupId].delay.withUpdate(newDelay, minSetback());
-        _groups[groupId].delay = updated;
+        uint48 effect;
+        (_groups[groupId].grantDelay, effect) = _groups[groupId].grantDelay.withUpdate(newDelay, minSetback());
```
Comment (frangio) [inline contracts/access/manager/AccessManager.sol:437]: Was this a bug?

File: contracts/access/manager/AccessManager.sol [hunk 6]
```diff
@@ -798,41 +752,51 @@ contract AccessManager is Context, Multicall, IAccessManager {
 
     /**
      * @dev Get the admin restrictions of a given function call based on the function and arguments involved.
+     *
+     * Returns:
+     * - bool enabled: does this data match a restricted operation
```
Comment (frangio) [inline contracts/access/manager/AccessManager.sol:757]:
```suggestion
     * - bool restricted: does this data match a restricted operation
```

File: contracts/access/manager/AccessManager.sol (hunk 4 above)
Comment (Amxx) [reply contracts/access/manager/AccessManager.sol:90]: done

**Pull request merged**